src/handlers.ts

```diff
@@ -45,7 +45,11 @@ export const takerRequestHandler = async (
             : quoter.fetchIndicativeQuoteAsync(takerRequest);
 
     const response = await responsePromise;
+    if (response && response.protocolVersion !== takerRequest.protocolVersion) {
+       throw new Error('Response and request protocol versions do not match');
```

Comment:
This is perfect, since we are in the web thread here I'm just going to finish things gracefully and return a `501 Not Implemented`